lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -138,14 +152,20 @@ public void collect(int doc) throws IOException {
           if (score > after.score || (score == after.score && doc <= afterDoc)) {
             // hit was collected on a previous page
             if (totalHitsRelation == TotalHits.Relation.EQUAL_TO && hitsThresholdChecker.isThresholdReached()) {
+              // Since the queue is prepopulated with sentinel objects, getting here means that the local
+              // priority queue is full
+              if (bottomValueChecker != null) {
```

Comment:
It would be easier to put this logic in updateMinCompetitiveScore as well ?

Reply:
Agreed -- The reason I did not put it in `updateMinCompetitiveScore` is because the function already became significantly heavy by the additions in this PR. Added now